### PR TITLE
 UNSPLASHAPP-010-Settings-Screen

### PR DESCRIPTION
**Pull request by IosBlessed**
In this ticket I've done: 
1. Created Settings screen using confirmed design from figma.
2. Involved interaction logic with buttons( log out, delete persistent data -> replace password must be done in next ticket).
3. Added alerts in case if user's tap can be defined as misclick.
4. Navigation between screen based on user's actions ( logout ).

### CODE REVIEW

File: unsplahAPI/unsplahAPI/Authentification/Authentication/View/AuthenticationViewController.swift [hunk 1]
```diff
@@ -65,7 +65,7 @@ final class AuthenticationViewController: UIViewController, AuthenticationViewCo
         viewModel?.isAuthenticated.bind { [weak self] isAuthenticated in
             guard let self else { return }
             if isAuthenticated {
-                self.coordinator?.didFinishAuthentification()
+                //self.coordinator?.didFinishAuthentification()
```
Comment (vikabelinschi) [inline unsplahAPI/unsplahAPI/Authentification/Authentication/View/AuthenticationViewController.swift:68]: Avoid commented code. If we don't need it, remove it. Else, uncomment it

Comment (IosBlessed) [reply unsplahAPI/unsplahAPI/Authentification/Authentication/View/AuthenticationViewController.swift:68]: this code is part of testing logic -> I used keychain for state of user, if I would like to test login flow or create account flow, I don't need to analyse if user already is kept there

File: unsplahAPI/unsplahAPI/Services/AppService/AppService.swift [hunk 2]
```diff
@@ -97,6 +97,46 @@ class UnsplashAPI {
...
+    // CRUD Keychain
+    
+    func getUserfromKeychain(completion: @escaping(PasswordDetails) -> Void) {
+        KeychainService.shared.getPassword { [weak self] passwordRequest in
+            guard let self else { return }
+            switch passwordRequest {
+            case .success(let passwordDetails):
+                return completion(passwordDetails)
+            case .failure(let error):
+                print(error)
```
Comment (vikabelinschi) [inline unsplahAPI/unsplahAPI/Services/AppService/AppService.swift:114]: Avoid prints

Comment (IosBlessed) [reply unsplahAPI/unsplahAPI/Services/AppService/AppService.swift:114]: it's for debug. We don't need them there as well?

File: unsplahAPI/unsplahAPI/Services/AppService/AppService.swift [hunk 3]
```diff
@@ -97,6 +97,46 @@ class UnsplashAPI {
...
+        }
+    }
+    
+    func saveUserDetailsToKeychain(username: String, password: String) {
+        KeychainService.shared.savePassword(username: username, password: password) { queryStatus in
+            switch queryStatus {
+            case .success(let successMessage):
+                print(successMessage)
+            case .failure(let error):
+                print(error)
```
Comment (vikabelinschi) [inline unsplahAPI/unsplahAPI/Services/AppService/AppService.swift:126]: Same here, in this case we can use it if we do not need any handling for the message, but usually we do not print data, we have some LogManager for logging that info.

**Review by vikabelinschi — APPROVED** [review]

File: unsplahAPI/unsplahAPI/Services/AppService/AppService.swift (hunk 3 above)
Comment (IosBlessed) [reply unsplahAPI/unsplahAPI/Services/AppService/AppService.swift:126]: I've got the idea, thanks for clarification

**Review by elanghel — APPROVED** [review]

**Pull request merged**